### PR TITLE
Make action names available in NodeClient

### PR DESCRIPTION
**Pull request by rjernst**
The actions available in NodeClient are registered on node startup and
hidden to callers. However, operator privileges needs to verify that all
actions are classified into operator and non-operator actions. Currently
the test uses reflection hacks to make the internal action objects
available.

This commit makes the action names available as a public method on
NodeClient, so that the reflection hacks are no longer necessary. It
would be nice to have a test specific way to expose this, but the test
code in question actually serves the action names up in a rest api, so
it is not test code, as far as server is concerned, that needs the
action names.

### CODE REVIEW

File: x-pack/plugin/security/qa/operator-privileges-tests/src/main/java/org/elasticsearch/xpack/security/operator/actions/RestGetActionsAction.java [hunk 1]
```diff
@@ -41,19 +33,7 @@ public String getName() {
     @SuppressWarnings({ "rawtypes", "unchecked" })
```
Comment (ywangd) [inline x-pack/plugin/security/qa/operator-privileges-tests/src/main/java/org/elasticsearch/xpack/security/operator/actions/RestGetActionsAction.java:33]: Both `@SuppressForbidden` and `@SuppressWarnings` do not seem needed with the proposed changes.

**Review by ywangd — APPROVED** [review]
LGTM

Thanks Ryan!

File: server/src/main/java/org/elasticsearch/client/internal/node/NodeClient.java [hunk 2]
```diff
@@ -67,6 +69,13 @@ public void initialize(
         this.namedWriteableRegistry = namedWriteableRegistry;
     }
 
+    /**
+     * Return the names of all available actions registered with this client.
+     */
+    public List<String> getActionNames() {
+        return actions.keySet().stream().map(ActionType::name).collect(Collectors.toList());
+    }
```
Comment (ChrisHegarty) [inline server/src/main/java/org/elasticsearch/client/internal/node/NodeClient.java:77]: Trivially, `.collect(Collectors.toList())` could be replaced with `.toList()`, which has the advantage of creating an unmodifiable list. (additionally, if this suggestions is adopted, then the javadoc could reflect the unmodifiability of the returned list)

**Review by ChrisHegarty — APPROVED** [review]

**Comment by rjernst** [comment]
@elasticmachine run elasticsearch-ci/bwc

**Pull request merged**